web/modules/custom/collection/src/ContextProvider/CollectionRouteContext.php

```diff
@@ -48,6 +51,22 @@ public function getRuntimeContexts(array $unqualified_context_ids) {
         }
       }
     }
+    elseif ($route_parameters = $this->routeMatch->getParameters()) {
+      /** @var ContentEntityInterface $route_parameter */
+      foreach ($route_parameters as $route_parameter) {
```

Comment:
I added a filter for the routes and added basic routes to the array.
Maybe an alter hook can be set there though to allow other modules provide their own routes.
Do you think it is needed?